targeting/geoip2/geoip2.go

```diff
@@ -123,6 +140,50 @@ func (g *GeoIP2) get(t geoType, db string) (*geoip2.Reader, error) {
 	return g.open(t, db)
 }
 
+func (g *GeoIP2) watchFiles() {
+	// Not worried about goroutines leaking because only one geoip2.New call is made in geodns.go (outside of testing)
+	ticker := time.NewTicker(1 * time.Minute)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				g.checkForUpdate()
+			default:
+				time.Sleep(12 * time.Second) // Sleep to avoid constant looping
+			}
+		}
+	}()
+}
+
+func (g *GeoIP2) checkForUpdate() {
+	// Iterate through each file, check modtime. If new, reload file
+	d := []*geodb{&g.country, &g.city, &g.asn} // Slice of pointers is kinda gross, but want to directly reference struct values (per const type)
```

Comment:
Would it be less "gross" if this was abstracted into a function that returns a list of the geodb's that are in use?

Reply:
Probably yes, but I worry about making "cleaner" code at the cost of performance or other integration/extension issues.   As I tried to convey in my comment, "kinda gross" is (to me) acceptable.